perma_web/frontend/components/UploadForm.vue

```diff
@@ -63,57 +46,61 @@ const handleClick = (e) => {
 }
 
 const handleUploadRequest = async () => {
-    if (!globalStore.isReady) {
+    if (captureStatus.value !== 'ready') {
         return
     }
 
-    handleErrorReset()
-    globalStore.captureStatus = "isValidating"
-
-    const csrf = getCookie("csrftoken")
-    const requestType = globalStore.captureGUID ? "PATCH" : "POST"
-    const requestUrl = "/archives/" + globalStore.captureGUID ? `${globalStore.captureGUID}/` : ""
+    // validation
+    errors.value = {}
+    if (!file.value) {
+        errors.value.file = ['File is required']
+    }
+    if (!props.captureGUID) {
+        if (!title.value) {
+            errors.value.title = ['Title is required']
+        }
+        if (!url.value) {
+            errors.value.url = ['URL is required']
+        }
+    }
+    if (Object.keys(errors.value).length > 0) {
+        return
+    }
 
+    // prepare form data
+    captureStatus.value = "isValidating"
     const formDataObj = new FormData();
     formDataObj.append('folder', globalStore.selectedFolder.folderId);
-
-    for (const [key, { value }] of Object.entries(formData.value)) {
-        formDataObj.append(key, value);
+    formDataObj.append('file', file.value);
+    if (!props.captureGUID) {
+        formDataObj.append('title', title.value);
+        formDataObj.append('description', description.value);
+        formDataObj.append('url', url.value);
     }
-
-    try {
-        const response = await fetch(requestUrl,
-            {
-                headers: {
-                    "X-CSRFToken": csrf,
-                },
-                method: requestType,
-                credentials: "same-origin",
-                body: formDataObj
-            })
-
-
-        if (!response?.ok) {
-            const errorResponse = await getErrorResponse(response)
-            throw errorResponse
-        }
-
-        const { guid } = await response.json()
-        globalStore.captureStatus = "success"
-
-        handleReset()
-
-        window.location.href = `${window.location.origin}/${guid}`
-    } catch (error) {
-        globalStore.captureError = "uploadError"
-        console.error("Upload request failed:", error);
-
-        if (error?.response) {
-            errors.value = error.response
+    
+    // send request
+    const requestType = props.captureGUID ? "PATCH" : "POST"
+    const requestUrl = "/archives/" + (props.captureGUID ? `${props.captureGUID}/` : "")
+    const { data, error, response } = await fetchDataOrError(requestUrl, {
+        method: requestType,
+        data: formDataObj,
+    })
+
+    // error handling
+    if (error) {
+        console.log(error, response)
+        if (data) {
+            errors.value = data
```

Comment:
Does this want to be `globalErrors.value`, like below, too?

Reply:
Theoretically no -- the idea here is the API would return `400` with data `{title: "Terrible title"}` and it would render the same way as the `validation` section previously in the function. Again I'm not sure what states would trigger that to test, though.

Reply:
Ah, maybe if you upload https://github.com/harvard-lil/perma-filecheck/blob/develop/test_assets/eicar-standard-antivirus-test-file-adobe-acrobat-attachment.pdf?